### PR TITLE
Covert NIC indicator api just perform with normal JSON as rest API

### PR DESCRIPTION
**Pull request by Rohit25negi**
Ministry has been so weird. The API which they requested to to be available with SOAP format. Now, they have changed their mind and want  normal JSON api.
Here is the ticket: https://app.asana.com/0/1133008090195679/1133442640656694/f

Please review this, as the priority this needs be live on Friday morning.

### CODE REVIEW

File: custom/icds_reports/views.py [hunk 1]
```diff
@@ -1773,7 +1773,7 @@ def valid_state_names(self):
 
 
 @location_safe
-@method_decorator([api_auth, csrf_exempt, toggles.ICDS_NIC_INDICATOR_API.required_decorator()], name='dispatch')
```
Comment (snopoke) [inline custom/icds_reports/views.py]: surely this should still be `csrf_exempt`

File: custom/icds_reports/views.py [hunk 2]
```diff
@@ -1773,7 +1773,7 @@ def valid_state_names(self):
...
+@method_decorator([api_auth, toggles.ICDS_NIC_INDICATOR_API.required_decorator()], name='dispatch')
```
Comment (snopoke) [inline custom/icds_reports/views.py]: is the toggle necessary?

**Review by snopoke — APPROVED** [review]

File: custom/icds_reports/views.py (hunk 2 above)
Comment (sravfeyn) [reply custom/icds_reports/views.py]: There is no other permission for these APIs that we can use to assign/deassign for mobile workers. Without this, all mobile workers could access the API.

Comment (sravfeyn) [reply custom/icds_reports/views.py]: Alternative is to create `API Access Permission` with list of APIs, that is similar to `View Report` permission.

**Comment by sravfeyn** [comment]
Looks good, pending Simon's comments.

File: custom/icds_reports/views.py (hunk 1 above)
Comment (Rohit25negi) [reply custom/icds_reports/views.py]: for get request also? is not it only required in POST? I had to put it just because of POST

**Pull request merged**